tools/utils/get-refs.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+const {execSync} = require('child_process');
+
+/** A regex to select a ref that matches our semver refs. */
+const semverRegex = /^(\d+)\.(\d)+\.x$/;
+
+/**
+ * Syncronously executes the command
+ *
+ * Return the trimmed stdout as a string, with an added attribute of the exit code.
+ */
+function exec(command) {
+  let output = new String();
+  try {
+    output += execSync(command,{stdio: ['pipe','pipe','ignore']}).toString().trim();
+    output.code = 0;
+  } catch (err) {
+    output.code = err.status || 1;
+  }
+  return output;
+}
+
+/**
+ * Sort a list of fullpath refs into a list and then provide the first entry.
+ *
+ * The sort order will first find master ref, and then any semver ref, followed
+ * by the rest of the refs in the order provided.
+ */
+function getRefFromBranchList(gitOutput) {
+  const branches = gitOutput.trim().split('\n').map(x => x.trim().split('/')[1]);
+  return branches.sort((a, b) => {
+    if (a === 'master') {
+      return -1;
+    }
+    if (b === 'master') {
+      return 1;
+    }
+    if (a.match(semverRegex) && b.match(semverRegex)) {
```

Comment:
Since you don't care about the captured groups:

```suggestion
    if (semverRegex.test(a) && semverRegex.test(b)) {
```